### PR TITLE
[SPARK-49201][PS][PYTHON][CONNECT] Reimplement `hist` plot with Spark SQL

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Reimplement `hist` plot with Spark SQL

### Why are the changes needed?
Reimplement `hist` plot with Spark SQL to support Spark Connect, since the `pyspark.ml.feature.Bucketizer` has not been supported with Spark Connect.

### Does this PR introduce _any_ user-facing change?
yes, follow plotting functions are enabled in Spark Connect:

- `{Frame, Series}.plot.hist`
- `{Frame, Series}.plot(kind="hist", ...)`

### How was this patch tested?
1, enabled parity tests;
2, manually check with:
```
import pyspark.pandas as ps
import pandas as pd
import numpy as np

np.random.seed(123)
df = pd.DataFrame(np.random.randint(1, 7, 6000), columns=['one'])
df['two'] = df['one'] + np.random.randint(1, 7, 6000)
df = ps.from_pandas(df)
df.plot.hist(bins=12, alpha=0.5)
```

before (Spark Classic):
image

after (Spark Connect):
image

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: python/pyspark/pandas/plot/core.py [hunk 1]
```diff
@@ -181,14 +182,31 @@ def compute_hist(psdf, bins):
         colnames = sdf.columns
         bucket_names = ["__{}_bucket".format(colname) for colname in colnames]
 
+        # TODO(SPARK-49202): register this function in scala side
```
Comment (zhengruifeng) [inline python/pyspark/pandas/plot/core.py:185]: will register this ML function in SparkSessionExtensions after https://github.com/apache/spark/pull/47637/files get merged

Comment (zhengruifeng) [reply python/pyspark/pandas/plot/core.py:185]: using a python udf for now

**Review by HyukjinKwon — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]
```
Run PYTHON_EXECUTABLE=python3.9 ./dev/lint-python
starting python compilation test...
python compilation succeeded.

starting black test...
black checks passed.

starting PySpark custom errors check...
PySpark custom errors check passed.

starting flake8 test...
flake8 checks failed:
./python/pyspark/pandas/plot/core.py:29:1: F401 'pyspark.sql.utils.is_remote' imported but unused
from pyspark.sql.utils import is_remote
^
1     F401 'pyspark.sql.utils.is_remote' imported but unused
1
1
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**